sdk/monitor/monitor-query/src/internal/modelConverters.ts

```diff
@@ -262,13 +263,10 @@ export function convertResponseForMetrics(
   // eslint-disable-next-line @typescript-eslint/no-unused-vars -- eslint doesn't recognize that the extracted variables are prefixed with '_' and are purposefully unused.
   const { resourceregion, value: _ignoredValue, interval, timespan, ...rest } = generatedResponse;
 
-  const obj: MetricsQueryResult = {
+  const obj: Omit<MetricsQueryResult, "getMetricByName"> = {
```

Comment:
NIT: Can't we use type inference here instead?

Reply:
I don't have a type defined though with just the properties of MetricsQueryResult

Reply:
That's ok, you can omit the type declaration and let typescript infer them (especially as your return value and the input to `createMetricsQueryResult` are typed). Up to you, I just personally like to keep type declarations to the edges (function entry and exit points) but it's totally a personal preference (hence a nit)

Reply:
I think i'll let these be - if i remove the type, TS complains on the lines after that - 
```
  if (resourceregion) {
    obj.resourceRegion = resourceregion;
  }
  if (interval) {
    obj.granularity = interval;
  }
  ```
  The error I get is that obj doesn't have these properties 😄